### PR TITLE
Sprint 23 TLT-1463 Adding the as_user_id parameter to accounts method functions

### PR DESCRIPTION
**Pull request by douglashall**
@eparker71 
@rascalking

### CODE REVIEW

File: canvas_sdk/methods/accounts.py [hunk 1]
```diff
@@ -19,8 +21,10 @@ def list_accounts(request_ctx, per_page=None, **request_kwargs):
         per_page = request_ctx.per_page
     path = '/v1/accounts'
     payload = {
-        'per_page' : per_page,
+        'per_page': per_page,
     }
+    if as_user_id:
+        payload['as_user_id'] = as_user_id
```
Comment (eparker71) [inline canvas_sdk/methods/accounts.py]: why not add this to the payload above, it's ok to be there if it's none. that will keep the same pattern as all the other modules.

File: canvas_sdk/methods/accounts.py [hunk 2]
```diff
@@ -69,9 +73,11 @@ def get_sub_accounts_of_account(request_ctx, account_id, recursive=None, per_pag
         per_page = request_ctx.per_page
     path = '/v1/accounts/{account_id}/sub_accounts'
     payload = {
-        'recursive' : recursive,
-        'per_page' : per_page,
+        'recursive': recursive,
+        'per_page': per_page,
     }
+    if as_user_id:
```
Comment (eparker71) [inline canvas_sdk/methods/accounts.py]: same as above

**Comment by eparker71** [comment]
You should add those methods in the static dir as well. Since this param can be added to any request, it should be included in the generate script. Until then, if it's in the static dir we'll know to copy those if the code gets regenerated.

**Comment by eparker71** [comment]
otherwise looks good :+1:

File: canvas_sdk/methods/accounts.py (hunk 1 above)
Comment (douglashall) [inline canvas_sdk/methods/accounts.py]: Will the underlying code not include 'as_user_id as a query parameter if it is set to None? The Canvas API endpoint will choke if as_user_id=None is included in the request.

Comment (eparker71) [inline canvas_sdk/methods/accounts.py]: hmm, not sure, all the other calls allow None. We should test it.

**Comment by douglashall** [comment]
@eparker71 Added modified account functions to static_methods. This should be good to go.

**Comment by douglashall** [comment]
This one is pointing to master. Moving to develop.

**Pull request closed without merging**